src/NextGen/identity/identity.ts

```diff
@@ -0,0 +1,31 @@
+import { MPID, User } from '@mparticle/web-sdk';
+import { IMParticleComponent } from '../component';
+import { IdentityCallback } from './identityCallback';
+
+// QUESTION: What is the diff between this and IdentityApiData?
+// TODO: sync this with android class
+export interface IdentityApiResult {}
+export interface IdentityApiRequest {}
+
+// QUESTION: Should we mirror the way Android handles HTTP codes, or do it the way we've been doing it?
+// https://docs.mparticle.com/developers/sdk/android/idsync/#idsync-status-codes
+// https://docs.mparticle.com/developers/sdk/web/idsync/#error-handling
```

Comment:
We should do a sync with Native to see where iOS's lives.  The more we can have in common, the better.  One improvement we need to make with web error handling is that there are results that are not http codes, https://github.com/mParticle/mparticle-web-sdk/blob/master/src/constants.ts#L151-L157

There are just error codes, which should live somewhere else as well.  It's unclear how these errors are represented (if at all) on native.

Reply:
Here's the iOS section on these error codes. https://docs.mparticle.com/developers/sdk/ios/idsync/#error-handling

iOS error handling is done very similarly to android by using a callback

Reply:
+1 on aligning implementations where we can